src/OpenFeature/Model/HookContext.cs

```diff
@@ -10,20 +10,22 @@ namespace OpenFeature.Model
     /// <seealso href="https://github.com/open-feature/spec/blob/v0.5.2/specification/sections/04-hooks.md#41-hook-context"/>
     public sealed class HookContext<T>
     {
+        private readonly SharedHookContext<T> _shared;

```

Comment:
Given the combination of evaluation context updates with before hooks, and mutable hook data per hook-instance, the number of potential copies increases. I moved most of the hook context into this shared component. Then the per instance contexts are just referencing the same data.

This also made it easier to not instantiate a extra HookData. It could be moved back, and the first instance used for the first hook as an alternative to this approach.